graspy/align/base.py

```diff
@@ -0,0 +1,127 @@
+            raise TypeError(msg)
+        # check for 2-dness and finiteness
+        X = check_array(X, copy=True)
+        Y = check_array(Y, copy=True)
+        # check for equal components
+        if X.shape[1] != Y.shape[1]:
+            msg = "Two datasets have different number of components!"
+            raise ValueError(msg)
+        return X, Y
+
+    @abstractmethod
+    def fit(self, X, Y):
+        """
+        Uses the two datasets to learn the matrix ``self.Q_`` that aligns the
+        first dataset with the second.
+
+        Parameters
+        ----------
+        X : np.ndarray, shape (n, d)
+            First dataset of vectors. These vectors need to have same number of
+            dimensions as ones in ``Y``, but the number of vectors can differ.
+
+        Y : np.ndarray, shape (m, d)
+            Second dataset of vectors. These vectors need to have same number
+            of dimensions as ones in ``X``, but the number of vectors can
+            differ.
+
+        Returns
+        -------
+        self : returns an instance of self
+        """
+        pass
+
+    def transform(self, X):
+        """
+        Transforms the dataset ``X`` using the learned matrix ``self.Q_``. This
+        may be the same as the first dataset as in .fit(), or a new dataset.
+        For example, additional samples from the same dataset.
+
+        Parameters
+        ----------
+        X : np.ndarray, shape (n, d)
+            Dataset of vectors. Needs to have the same number of dimensions as
+            ``X`` and ``Y`` passed to fit, but can have a different number of
+            entries.
+
+        Returns
+        -------
+        X_prime : np.ndarray, shape (n, d)
+            First dataset of vectors, aligned to second. Equal to
+            ``X`` @ ``self.Q_``.
+        """
+        if not isinstance(X, np.ndarray):
+            msg = f"Dataset is a {type(X)}, not an np.ndarray! "
+            raise TypeError(msg)
+        X = check_array(X)
+        if not X.shape[1] == self.Q_.shape[0]:
+            msg = (
+                "Dataset needs to have the same number of dimensions, d, "
+                "as datasets X and Y used in fit. Currently, vectors in "
+                f"the dataset to transform have {X.shape[1]} dimensions, "
+                f"while vectors in fit had {self.Q_.shape[0]} dimensions."
+            )
+            raise ValueError(msg)
+        return X @ self.Q_
+
+    def fit_transform(self, X, Y):
+        """
+        Transforms the dataset ``X`` using the learned matrix ``self.Q_``. This
+        may be the same as the first dataset as in .fit(), or a new dataset.
+        For example, additional samples from the same dataset.
+
+        Parameters
+        ----------
+        X : np.ndarray, shape (n, d)
+            First dataset of vectors. These vectors need to have same number of
+            dimensions as ones in ``Y``, but the number of vectors can differ.
```

Comment:
@bdpedigo do you want an overloaded fit transform for orthogonal procrustes without `but the number of vectors can differ.`?

Reply:
nevermind, did it via overloading and calling super()